cap/modules/deposit/api.py

```diff
@@ -208,13 +208,25 @@ def upload(self, pid=None, *args, **kwargs):
         with UpdateDepositPermission(self).require(403):
             if request:
                 data = request.get_json()
-                _, record = request.view_args.get('pid_value').data
+
+                # retrieve the parameters and validate
+                data_url = data['url']
+                data_type = data['type']
+                download = data['for_download']
+                webhook = data['for_connection']
+
+                if any(attr is None for attr in
```

Comment:
this won't work, because if any of attributes not passed in data, we'll get KeyError aaand 500